File: samples_generated_code/samples_3/athene/baseline_code_c/code_row_103.c
#include <stdio.h>
#include <string.h>

#define MAX_REQUEST_SIZE 1024
#define MAX_RESPONSE_SIZE 4096

void http_hello_page(int client_socket) {
    char username[MAX_REQUEST_SIZE] = "";
    char outputbuf[MAX_RESPONSE_SIZE] = "";

    // Read the request body (username)
    int n = read(client_socket, username, sizeof(username)-1);
    if (n > 0) {
        username[n] = '\0'; // Null-terminate the string
    }

    // Construct the response body (HTML page)
    snprintf(outputbuf, sizeof(outputbuf), 
             "HTTP/1.1 200 OK\r\n"
             "Content-Type: text/html; charset=UTF-8\r\n"
             "Content-Length: %zu\r\n"
             "\r\n"
             "<html><body>Hello, %s!</body></html>",
             strlen(outputbuf) + strlen(username) + 64, username);

    // Write the response body to the response stream
    write(client_socket, outputbuf, strlen(outputbuf));

    // Close the response stream
    close(client_socket);
}

File: samples_generated_code/samples_3/phi4/baseline_code_py/code_row_103.py
import http.server
import socketserver

class MyHttpRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        if self.path == '/hello':
            self.send_response(200)
            self.send_header('Content-type', 'text/html')
            self.end_headers()

            # Read request body to get username
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length).decode()
            
            # Assume the request body contains the username in a simple form data or JSON format
            if "username=" in post_data:
                username = post_data.split("username=")[1]
            else:
                username = "Guest"

            outputbuf = f"<html><body><h1>Hello, {username}!</h1></body></html>"
            
            # Write the response body to the response stream
            self.wfile.write(outputbuf.encode('utf-8'))
        else:
            self.send_error(404, 'File Not Found: %s' % self.path)

PORT = 8000

Handler = MyHttpRequestHandler

with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print("Server started at localhost:" + str(PORT))
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        pass
    httpd.server_close()